src/client_lib/pegasus_client_impl.h

```diff
@@ -221,6 +221,10 @@ class pegasus_client_impl : public pegasus_client
                                  const scan_options &options,
                                  async_get_unordered_scanners_callback_t &&callback) override;
 
+    virtual bool detect_hotkey(const std::string &hotkey_type,
```

Comment:
Better to use dsn::apps::hotkey_type::type and dsn::apps::hotkey_detect_action as parameter, then no need to parse from string.